service/src/main/java/bio/terra/workspace/common/utils/FlightUtils.java

```diff
@@ -220,6 +220,11 @@ public static FlightState waitForJobFlightCompletion(Stairway stairway, String f
    */
   public static StepResult waitForSubflightCompletion(Stairway stairway, String flightId)
       throws InterruptedException {
+    return waitForSubflightCompletionAndReturnFlightInfo(stairway, flightId).getStepResult();
+  }
+
+  public static StepResultWithFlightInfo waitForSubflightCompletionAndReturnFlightInfo(
```

Comment:
🤔 so the tradeoff with the shift to this method is that we won't get the ability to customize the timeouts at the callsites and instead enforce a universal 60 minute timeout. Scanning through the changes I don't see any timeouts that will be lowered.

What do you think about getting a unit test around this code? I'm not sure we have any and it'd be good to get regression coverage on the retry and completion logic to avoid situations that caused this ticket (i.e., misinterpreting QUEUED as a terminal state).

Reply:
If we wish to customize the timeouts, we could add the ability to pass in those parameters. I started to do that, but then decided we should just wait until we decide it's actually needed.

I'll see about adding some unit test coverage for the `waitForSubflightCompletionAndReturnFlightInfo ` utility method.

Reply:
@aherbst-broad I have added a test and did some refactoring per Marc's suggestion (https://github.com/DataBiosphere/terra-workspace-manager/pull/1688/commits/d12eff57e3d8bb750c882d5e6abc250afd00b6fc). I'm going to test in a bee, but this is ready for review again.